### PR TITLE
Feature bl 8688 component recalls

### PR DESCRIPTION
**Pull request by ryszardm-kainos**

### CODE REVIEW

File: database/scripts/loadDevData.js [hunk 1]
```diff
@@ -130,33 +177,126 @@ function insertMakesToDb() {
...
+            getRecallCount(countCallback);
+          }, 1000);
+        });
+      }
+    }
+  };
+};
+
+// Increase dynamoDB throughput to load data faster, and start the load via throughputCallback
+dynamoDB.updateTable(getTableThroughputParams(RECALLS_TABLE_NAME, UPSACLED_THROUGHPUT, RECALLS_TABLE_SECONDARY_INDEX), function(err, data) {
```
Comment (zakrzewskam) [inline database/scripts/loadDevData.js]: Does ReadCapacityUnits need to be as big as WriteCapacityUnits?

File: frontend/controllers/modelController.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
-const url = require('url');
 const recallSearch = require('../service/recallSearch');
 const modelValidator = require('../validators/vehicleModel');
 const SmartSurveyFeedback = require('../helpers/SmartSurveyFeedback');
 
+const NOTICE_VEHICLE = 'This service only includes vehicles that have been recalled.';
```
Comment (zakrzewskam) [inline frontend/controllers/modelController.js]: Why are we setting this information in more that one place (makeController, modelController)?

File: frontend/controllers/modelController.js [hunk 3]
```diff
@@ -16,10 +18,12 @@ class ModelController {
       if (err) {
         console.error(err);
       } else {
+        const recallsAvailabilityNotice = recallType === 'vehicle' ? NOTICE_VEHICLE : NOTICE_EQUIPMENT;
```
Comment (zakrzewskam) [inline frontend/controllers/modelController.js]: Why recalls type is not a variable / enum somewhere?

File: frontend/routes/index.js [hunk 4]
```diff
@@ -20,67 +20,70 @@ router.get('/terms-and-conditions', (req, response) => {
...
-router.get('/vehicle-make', (req, response) => {
-  const recallType = req.query.recallType;
+router.get('/recall-types/:recallType/makes', (req, response) => {
+  const recallType = req.params.recallType;
   makeController.makesList(null, response, recallType);
 });
 
-router.post('/vehicle-make', (req, response) => {
-  const recallType = req.query.recallType;
+router.post('/recall-types/:recallType/makes', (req, response) => {
```
Comment (zakrzewskam) [inline frontend/routes/index.js]: Url suggests that you can post with more than one recall-types, makes, models, etc.

File: selenium/src/main/java/uk/gov/dvsa/recalls/ui/page/ResultsPage.java [hunk 5]
```diff
@@ -68,7 +68,7 @@ public boolean isHowToCheckSentenceValid(String recallType) {
 
     public boolean isNumberOfAffectedVehiclesValid(String recallType) {
```
Comment (zakrzewskam) [inline selenium/src/main/java/uk/gov/dvsa/recalls/ui/page/ResultsPage.java]: This check is not valid for plural form (contains not equal).

File: frontend/service/pluralForm.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ class PluralForm {
     "There is 1 recall."
 */
   static getSingularOrPlural(text, number) {
-    const indx = number > 1 ? 0 : 1;
+    const indx = number !== 1 ? 0 : 1;
```
Comment (zakrzewskam) [inline frontend/service/pluralForm.js]: You need to update description after this change

File: frontend/routes/index.js (hunk 4 above)
Comment (ryszardm-kainos) [reply frontend/routes/index.js]: Oh joy... :D

**Pull request merged**